haystack/nodes/answer_generator/transformers.py

```diff
@@ -413,6 +413,12 @@ def _register_converters(cls, model_name_or_path: str, custom_converter: Optiona
 
     @classmethod
     def _get_converter(cls, model_name_or_path: str) -> Optional[Callable]:
+        # using dictionary key based on model_name_or_path endswith
+        if model_name_or_path.endswith("bart_lfqa"):
+            return cls._model_input_converters.get("vblagoje/bart_lfqa")
+        elif model_name_or_path.endswith("bart_eli5"):
+            return cls._model_input_converters.get("yjernite/bart_eli5")
+        # added original return statement for linting
         return cls._model_input_converters.get(model_name_or_path)
```

Comment:
sorry for intruding...

I think the logic here can be improved.
My impression is that while using standard models, the last line is never called

What about changing `model_name_or_path` inside the method?

Something like this (untested):

```python
    def _get_converter(cls, model_name_or_path: str) -> Optional[Callable]:
        # using dictionary key based on model_name_or_path endswith
        if model_name_or_path.endswith("bart_lfqa"):
            model_name_or_path="vblagoje/bart_lfqa"
        elif model_name_or_path.endswith("bart_eli5"):
             model_name_or_path="yjernite/bart_eli5"
        return cls._model_input_converters.get(model_name_or_path)
```

Or maybe there are also better solutions... 🤔

Reply:
@anakin87 Agreed, this is better. I was not sure if I should change what the function returns solely to what's inside my added loop so I left the return statement outside as well.